contracts/validator-manager/tests/NativeTokenStakingManagerTests.t.sol

```diff
@@ -249,25 +250,15 @@ contract NativeTokenStakingManagerTest is PoSValidatorManagerTest {
     }
 
     function _expectRewardIssuance(address account, uint256 amount) internal override {
-        vm.mockCall(
-            address(app.NATIVE_MINTER()),
-            abi.encodeCall(INativeMinter.mintNativeCoin, (account, amount)),
-            ""
-        );
-        // empty calldata implies the receive function will be called:
-        vm.mockCall({
-            callee: account,
-            msgValue: amount,
-            data: "", // implies receive()
-            returnData: ""
-        });
-        // Units tests don't have access to the native minter precompile, so use vm.deal instead.
-        vm.deal(account, account.balance + amount);
+        address nativeMinter = address(app.NATIVE_MINTER());
+        bytes memory callData = abi.encodeCall(INativeMinter.mintNativeCoin, (account, amount));
+        vm.mockCall(nativeMinter, callData, "");
+        vm.expectCall(nativeMinter, callData);
     }
 
     function _setUp() internal override returns (IValidatorManager) {
         // Construct the object under test
-        app = new NativeTokenStakingManager(ICMInitializable.Allowed);
+        app = new MockNativeTokenStakingManager(ICMInitializable.Allowed);
```

Comment:
I'm not sure that `MockNativeTokenStakingManager` is the best name here since you aren't really mocking here but monkeypatching instead. How about `TestableNativeTokenStakingManager`

Reply:
done